mcs/class/System/Mono.Net.Security/MobileAuthenticatedStream.cs

```diff
@@ -338,6 +338,26 @@ public override Task WriteAsync (byte[] buffer, int offset, int count, Cancellat
 			return StartOperation (OperationType.Write, asyncRequest, cancellationToken);
 		}
 
+		public override IAsyncResult BeginRead (byte[] buffer, int offset, int count, AsyncCallback callback, object state)
+		{
+			return TaskToApm.Begin (ReadAsync (buffer, offset, count), callback, state);
+		}
+
+		public override int EndRead (IAsyncResult asyncResult)
+		{
+			return TaskToApm.End<int> (asyncResult);
```

Comment:
Please do not put any `TaskToApm` usage into this class.  This `EndRead()` method will not be called after the `SslStream` has been disposed - thus leaking an unobserved task.